File: Project/C/Moviesion01/ca.uwindsor.cs.comp3150.pl/Moviesion.c
#include <stdio.h>
#include <string.h>

#include "../ca.uwindsor.cs.comp3150.cmn/Director.c"
#include "../ca.uwindsor.cs.comp3150.cmn/Movie.c"

#define DBSIZE 100

void addMovie(int movieId, struct Movie movies[]) {
	
	struct Movie newMovie;	
	newMovie.id = movieId;
	printf("Enter movie title:\n");	
	scanf("%s", newMovie.title);

	struct Director newDirector;
	printf("Enter director name:\n");
	scanf("%s", newDirector.name);
	
	//Relationship between movie and director
	strcpy(newMovie.director.name, newDirector.name);
	
	//Add to our in-memory movie database
	movies[movieId] = newMovie;
	printf("Movie '%s' has been added with Id='%d'\n", newMovie.title, newMovie.id);	
}

void printMovieByName(struct Movie movies[]) {
	//find the movie by its name and print out its full info.
}

int printMovieById(struct Movie movies[]) {

	printf("Enter movie id:\n");
	int id;
	scanf("%d", &id);
	for(int i=0; i < DBSIZE; ++i){
		if(movies[i].id == id){
			printf("Title: %s\nDirector: %s\n", movies[i].title, movies[i].director.name);
			return id;
		}
	}
	printf("Movie not found!\n");
	return -1;
}

void printAllMoviesByDirector(struct Movie movies[]) {
	printf("Enter director name: \n");
	char directorName[100];
	scanf("%s", directorName);
	for(int i=0; i < DBSIZE; ++i){
		if(movies[i].id > 0 && strcmp(movies[i].director.name, directorName) == 0){
			printf("Movie title: %s\n", movies[i].title);
		}                
	}
}

void printAllMovies(struct Movie movies[]) {
	//loop over all movies and print out the info.
}

void editMovieById(struct Movie movies[]) {
	
	int id = printMovieById(movies);
	if(id < 0) return;
	
	printf("Enter movie new title:\n");	
	scanf("%s", movies[id].title);
	
	printf("Enter new director name: \n");	
	scanf("%s", movies[id].director.name);
	
	printf("Movie with Id='%d' has been updated", id);
}

void deleteMovieById(struct Movie movies[]) {
	//find the movie by Id and remove it from array
}

void main() {

	//Our in-memory movie database
	struct Movie movies[DBSIZE];
	
	//Auto increment id
	int movieId = 1;
	
	char* commandList = 
			"1) Add Movie\n\
2) Print Movie by Id\n\
3) *Print Movie by Title\n\
4) *Print All Movies\n\
5) Print All Movies of a Director\n\
6) Edit Movie by Id\n\
7) *Delete Movie by Id\n\
0) Quit\n\
Enter command: ";
	int running = 1;
	int commandId;
	while (running) {		
		printf(commandList);
		scanf("%d", &commandId);
		switch (commandId) {
			case -1:
				break;
			case 0:
				printf("Bye.\n");
				running = 0;
				break;
			case 1:
				addMovie(movieId, movies);
				movieId++;
				break;
			case 2:
				printMovieById(movies);
				break;
			case 3:
				printMovieByName(movies);
				break;
			case 4:
				printAllMovies(movies);
				break;
			case 5:
				printAllMoviesByDirector(movies);
				break;
			case 6:
				editMovieById(movies);
				break;
			case 7:
				deleteMovieById(movies);
				break;
			default:
				printf("Command not recognized!");
				break;		
		}
		printf("\n\n");
	
	}
}

File: Project/Java/Moviesion01/src/ca/ryerson/ee/coe848/pl/Moviesion.java
package ca.ryerson.ee.coe848.pl;

import ca.ryerson.ee.coe848.cmn.Director;
import ca.ryerson.ee.coe848.cmn.Movie;
import java.util.ArrayList;
import java.util.Scanner;

public class Moviesion {
    public static void main(String[] args) {

        //Our in-memory movie database
        ArrayList<Movie> movies = new ArrayList<Movie>();
        
        //Auto increment id
        Integer movieId = 100;
        
        Scanner console = new Scanner(System.in);
        String commandList = 
                  "1) Add Movie\n"
                + "2) Print Movie by Id\n"
                + "3) *Print Movie by Title\n"
                + "4) *Print All Movies\n"
                + "5) Print All Movies of a Director\n"
                + "6) *Edit Movie by Id\n"
                + "7) Delete Movie by Id\n"
                + "0) Quit\n"
                + "Enter command: ";
        System.out.println(commandList);
        boolean running = true;
        Integer commandId;
        while (running) {
            try {
                try {
                    commandId = Integer.parseInt(console.nextLine());
                } catch (Exception ex) {
                    System.out.println("Command not recognized!");
                    commandId = -1;
                }
                switch (commandId) {

                    case -1:
                        break;
                    case 0:
                        System.out.println("Bye.");
                        running = false;
                        break;
                    case 1:
                        addMovie(movieId, movies, console);
                        movieId++;
                        break;
                    case 2:
                        printMovieById(movies, console);
                        break;
                    case 3:
                        printMovieByName(movies, console);
                        break;
                    case 4:
                        printAllMovies(movies, console);
                        break;
                    case 5:
                        printAllMoviesByDirector(movies, console);
                        break;
                    case 6:
                        editMovieById(movies, console);
                        break;
                    case 7:
                        deleteMovieById(movies, console);
                        break;
                    default:
                        System.out.println("Command not recognized!");
                        break;
                }
                System.out.println(commandList);
            } catch (Exception ex) {
                System.out.println("Command failed!\n" + ex.getMessage());
                System.out.println(commandList);
            }
        }
        console.close();
    }

    private static void addMovie(Integer movieId, ArrayList<Movie> movies, Scanner console) {
        
        System.out.println("Enter movie title:");
        String movieTitle = console.nextLine();
        Movie newMovie = new Movie();
        newMovie.setTitle(movieTitle);
        newMovie.setId(movieId);
        
        System.out.println("Enter director name: ");
        String directorName = console.nextLine();
        Director newDirector = new Director();
        newDirector.setName(directorName);
        
        //Relationship between movie and director
        newMovie.setDirector(newDirector);
        
        //Add to our in-memory movie database
        movies.add(newMovie);
        System.out.println(
                String.format("Movie '%s' has been added with Id='%s'", 
                        newMovie.getTitle(), 
                        newMovie.getId()));
        
    }

    private static void printMovieByName(ArrayList<Movie> movies, Scanner console) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    private static void printMovieById(ArrayList<Movie> movies, Scanner console) {

        System.out.println("Enter movie id:");
        Integer id;
        try {
            id = Integer.parseInt(console.nextLine());
        } catch (Exception ex) {
            System.out.println("Movie id not recognized!");
            return;
        }
        
        for(Movie m: movies){
            if(m.getId() == id){
                System.out.println(m.toString());
                return;
            }
        }
        System.out.println("Movie not found!");
    }
    
    private static void printAllMovies(ArrayList<Movie> movies, Scanner console) {
        throw new UnsupportedOperationException("Not supported yet."); 
    }

    private static void printAllMoviesByDirector(ArrayList<Movie> movies, Scanner console) {
        System.out.println("Enter director name: ");
        String dircetorName = console.nextLine();
        for(Movie m: movies){
            if(m.getDirector().getName().equalsIgnoreCase(dircetorName)){
                System.out.println(m.toString());
            }                
        }
    }

    private static void editMovieById(ArrayList<Movie> movies, Scanner console) {
        throw new UnsupportedOperationException("Not supported yet."); 
    }

    private static void deleteMovieById(ArrayList<Movie> movies, Scanner console) {
        System.out.println("Enter movie id:");
        Integer id;
        try {
            id = Integer.parseInt(console.nextLine());
        } catch (Exception ex) {
            System.out.println("Movie id not recognized!");
            return;
        }
        
        for(Movie m: movies){
            if(m.getId() == id){
                movies.remove(m);
                System.out.println("Movie is removed.");
                return;
            }
        }
        System.out.println("Movie not found!");
    }
}
